src/internal/utils/scrollable-containers.ts

```diff
@@ -84,3 +84,28 @@ export function scrollElementIntoView(
 ) {
   element?.scrollIntoView?.(options);
 }
+
+export function calculateScroll({ top, height }: BoundingBox) {
+  if (top < 0) {
+    return top;
+  } else if (top + height > window.innerHeight) {
+    if (height > window.innerHeight) {
+      return top;
+    } else {
+      return top + height - window.innerHeight;
+    }
+  }
+  return 0;
+}
+
+/**
+ * For elements with fixed position, the browser's native scrollIntoView API doesn't work,
+ * so we need to manually scroll to the element's position.
+ * Supports only vertical scrolling.
+ */
+export function scrollRectangleIntoView(box: BoundingBox) {
+  const scrollAmount = calculateScroll(box);
+  if (scrollAmount) {
+    window.scrollBy(0, scrollAmount);
```

Comment:
What if the chart displayed inside scrollable container? For example a split panel? Do we need to support this?

Reply:
Is this what you mean? It works as usual with the caveat that the potentially invisible part of the chart leads to an apparent white space between it and the popover (the second screenshot below makes this more apparent, because of the chart's scrolling position). This could be mitigated by scrolling the scrolling container, but I'm not sure we want the extra complexity.

Screenshot 1: chart more or less centered in its scrollable container

![localhost_8080_ (48)](https://github.com/cloudscape-design/components/assets/1257272/3433784a-40fb-4155-a62a-3bee50d768d8)

Screenshot 2: chart at the bottom, leaving most of its upper part cropped

![localhost_8080_ (49)](https://github.com/cloudscape-design/components/assets/1257272/2eeb2e79-f9bc-4e3a-b63d-6161a3f0e11a)